src/app/common/services/gateways/gateways.service.ts

```diff
@@ -49,64 +50,91 @@ export class GatewaysService {
+                  metadata: {
+                    type: this.typeDataChan,
+                  },
+                };
+                this.channelsService.addChannel(dataChannel).subscribe(
+                  respAddData => {
+                    const dataChanID = respAddData.headers.get('location').replace('/channels/', '');
+
+                    const exportChannel: Channel = {
+                      name: `${gateway.name}-${this.typeExportChan}`,
+                      metadata: {
+                        type: this.typeExportChan,
+                      },
+                    };
+                    this.channelsService.addChannel(exportChannel).subscribe(
+                      respAddExport => {
+                        const exportChanID = respAddExport.headers.get('location').replace('/channels/', '');
+
+                        this.channelsService.connectThing(ctrlChanID, gwID).subscribe(
+                          respConnectCtrl => {
+                            this.channelsService.connectThing(dataChanID, gwID).subscribe(
+                              respConnectData => {
+                                this.channelsService.connectThing(exportChanID, gwID).subscribe(
+                                  respConnectExport => {
+                                    gateway.metadata.ctrlChannelID = ctrlChanID;
+                                    gateway.metadata.dataChannelID = dataChanID;
+                                    gateway.metadata.exportChannelID = exportChanID;
+                                    gateway.metadata.gwPassword = uuid();
+                                    gateway.id = gwID;
+
+                                    this.thingsService.editThing(gateway).subscribe(
+                                      () => {
+                                        this.notificationsService.success('Gateway successfully created', '');
+
+                                        // Send fake location
+                                        this.messagesService.sendLocationMock(dataChanID, gwID);
+
+                                        // Bootstrap gateway
+                                        this.bootstrapService.addConfig(gateway).subscribe();
+                                      },
+                                      errEdit => {
+                                        this.deleteGateway(gateway).subscribe();
+                                      },
+                                    );
+                                  },
+                                  errExportConnect => {
+                                    this.deleteGateway(gateway).subscribe();
+                                  },
+                                );
+                              },
+                              errDataConnect => {
+                                this.deleteGateway(gateway).subscribe();
+                              },
+                            );
```

Comment:
Maybe you should think of making subroutines, so that this nesting looks a bit tidier. Dunno :)